numpy/lib/_function_base_impl.py

```diff
@@ -5178,14 +5179,9 @@ def delete(arr, obj, axis=None):
     array([ 1,  3,  5,  7,  8,  9, 10, 11, 12])
 
     """
-    wrap = None
-    if type(arr) is not ndarray:
-        try:
-            wrap = arr.__array_wrap__
-        except AttributeError:
-            pass
+    conv = _array_converter(arr)
+    arr, = conv.as_arrays(subok=False)
```

Comment:
Why this had `asarray` rather than `asanyarray` -- o well, here definitely ensure no change in API!

Though perhaps still do `arr = conv.as_arrays(subok=False)[0]`?

Reply:
OK, this is clearly in the eye of the beholder!